Package.swift

```diff
@@ -17,6 +17,10 @@ let package = Package(
         .library(name: "AWSLambdaRuntimeCore", targets: ["AWSLambdaRuntimeCore"]),
         // plugin to package the lambda, creating an archive that can be uploaded to AWS
         .plugin(name: "AWSLambdaPackager", targets: ["AWSLambdaPackager"]),
+        // plugin to deploy the lambda, relies on AWS SAM command line
+        .plugin(name: "AWSLambdaDeployer", targets: ["AWSLambdaDeployer"]),
+        // Shared Library to generate a SAM deployment descriptor
+        .library(name: "AWSLambdaDeploymentDescriptor", type: .dynamic, targets: ["AWSLambdaDeploymentDescriptor"]),
```

Comment:
what role is `type: .dynamic` playing here?

Reply:
The plugin calls a bash shell script.
The bash shell script calls `swift Deploy.swift` to dynamically compile and execute the deployment descriptor definition. 
`Deploy.swift` has a dependency on the data structure and the code to generate the deployment descriptor.
The only way I found to load this dependency is using a dynamic library.  

The bash shell script calls `swift -lsharedLib` to load the shared library, as seen in comment https://github.com/swift-server/swift-aws-lambda-runtime/pull/291#discussion_r1132769776

Swift Package's manager do not need this step because `PackageDefinition` is part of the language runtime and correctly installed at the time `swift` command is called.  

Here, with the plugin, the shared lib is in the `.build` directory of the swift-lambda-runtime project.

Let me know if you think there is a different way to make the deployment descriptor code available to `Deploy.swift`